tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -308,6 +308,21 @@ class Restrictions {
       ];
     }
 
+    var currentModel =
+        parsedModels.findByClassName(documentDefinition!.className);
+
+    var ancestorServerOnlyClass =
+        _findServerOnlyClassInParentClasses(currentModel);
+
+    if (!documentDefinition!.serverOnly && ancestorServerOnlyClass != null) {
+      return [
+        SourceSpanSeverityException(
+          'The class "${documentDefinition!.className}" cannot inherit from the server-only class "${ancestorServerOnlyClass.className}" unless it is also marked as "serverOnly".',
```

Comment:
Suggestion: How about `Cannot extend a "serverOnly" class in the inheritance chain ("${ancestorServerOnlyClass.className}") unless class is marked as "serverOnly".`?